### ISSUE TITLE
chmod: cannot access 'bin/magento': No such file or directory

### ISSUE THREAD

**Issue by jaywtc** [markshust/docker-magento#731]
When installing Magent0 2.4.3 p1  

setup like bin/setup magento.test

it showing this error

**Comment by LiamKarlMitchell** [comment]
Work-around on WSL 2 Ubuntu for me.

1. `docker exec -it magento-phpfpm-1 bash`
2. `composer create-project --repository-url=https://repo.magento.com/ magento/project-community-edition=2.4.3 .`
3. `mv project-community-edition/* .`
    Note: Not sure if needed, I may have missed the . off the end of the composer install.
4. `exit`
5. `bin/magento`

**Comment by markshust** [comment]
Thanks for posting this info. I'll try to finally do an install of docker-magento on Linux after I tag & publish the next version to make sure the README is still correct. Definitely willing to take in any updates to it if needed.

**Comment by LiamKarlMitchell** [comment]
Thanks Mark, if you could drop a message here when the next version is up, perhaps I can try it on weekend :)

**Comment by myselfhimself** [comment]
Reproducing bug for 2.4.3-p2 with `curl -s https://raw.githubusercontent.com/markshust/docker-magento/master/lib/onelinesetup | bash -s -- magento.test 2.4.3-p2`. Could not reproduce fix...

**Comment by Asitis** [comment]
@markshust I moved to the 2.4.4 install myself because that works fine out of the box - but I believe there is something off when loading Magento into the docker file, possibly related to permission errors that are not reproducable on macOS, [as read here](https://www.fullstaq.com/knowledge-hub/blogs/docker-and-the-host-filesystem-owner-matching-problem).
With my 2.4.4 install I also had some permission issues, that I got fixed after reading that article.

For me this happened on Ubuntu 22.04 LTS, fresh install.

**Comment by myselfhimself** [comment]
> @myselfhimself can you please post your full operating system & host details? Everything works fine for me on macOS, so it'll be very hard/impossible to confirm this is even an issue without all of the debugging details.

```
Micro-Star International Co., Ltd. MS-7D25
32,0 GiB
12th Gen Intel® Core™ i7-12700K × 20
Ubuntu 22.04 LTS
64-bit
--
Docker version 20.10.17, build 100c701
```

**Comment by markshust** [comment]
Are there any errors occurring when running `bin/setup`? If so and you can post the full output, we can try to find out what is going on. Manually cloning out the project within the Docker container shouldn't be needed.

**Comment by myselfhimself** [comment]
> Are there any errors occurring when running `bin/setup`? If so and you can post the full output, we can try to find out what is going on. Manually cloning out the project within the Docker container shouldn't be needed.

See attachment
[magento-2.4.3-p2-install-setup-errors.txt](https://github.com/markshust/docker-magento/files/9149879/magento-2.4.3-p2-install-setup-errors.txt)

**Comment by markshust** [comment]
@myselfhimself appreciate it.

It appears this issue is related to Composer:

```
  - Installing dealerdirect/phpcodesniffer-composer-installer (v0.7.2): Extracting archive
dealerdirect/phpcodesniffer-composer-installer contains a Composer plugin which is blocked by your allow-plugins config. You may add it to the list if you consider it safe. See https://getcomposer.org/allow-plugins
You can run "composer config --no-plugins allow-plugins.dealerdirect/phpcodesniffer-composer-installer [true|false]" to enable it (true) or keep it disabled and suppress this warning (false)
  - Installing magento/composer-dependency-version-audit-plugin (0.1.1): Extracting archive
magento/composer-dependency-version-audit-plugin contains a Composer plugin which is blocked by your allow-plugins config. You may add it to the list if you consider it safe. See https://getcomposer.org/allow-plugins
You can run "composer config --no-plugins allow-plugins.magento/composer-dependency-version-audit-plugin [true|false]" to enable it (true) or keep it disabled and suppress this warning (false)
  - Installing magento/composer-root-update-plugin (2.0.2): Extracting archive
magento/composer-root-update-plugin contains a Composer plugin which is blocked by your allow-plugins config. You may add it to the list if you consider it safe. See https://getcomposer.org/allow-plugins
You can run "composer config --no-plugins allow-plugins.magento/composer-root-update-plugin [true|false]" to enable it (true) or keep it disabled and suppress this warning (false)
  - Installing magento/inventory-composer-installer (1.2.0): Extracting archive
magento/inventory-composer-installer contains a Composer plugin which is blocked by your allow-plugins config. You may add it to the list if you consider it safe. See https://getcomposer.org/allow-plugins
You can run "composer config --no-plugins allow-plugins.magento/inventory-composer-installer [true|false]" to enable it (true) or keep it disabled and suppress this warning (false)
  - Installing magento/magento-composer-installer (0.3.0): Extracting archive
magento/magento-composer-installer contains a Composer plugin which is blocked by your allow-plugins config. You may add it to the list if you consider it safe. See https://getcomposer.org/allow-plugins
You can run "composer config --no-plugins allow-plugins.magento/magento-composer-installer [true|false]" to enable it (true) or keep it disabled and suppress this warning (false)
```

...and can be solved with: https://magento.stackexchange.com/questions/352353/why-composer-install-is-asking-to-allow-plugins

So, I think we just need to update the Linux README to include this step in the process? If someone on Linux can verify it, that would be extremely helpful. Here's a link to the Linux part of the readme: https://github.com/markshust/docker-magento#linux

**Comment by myselfhimself** [comment]
@markshust The wise paths to fix in order to keep the docker-magento oneliner install script mostly automated would be to ask Magento 2's upstream project to add that composer.json fix to Magento <2.4.4 releases  (asking now) or to induce the `composer` executable call to non-interactively accept all unsafe dependencies with some --force flag if that exists.

**Comment by myselfhimself** [comment]
@markshust On the non-interactive `composer` call strategy, according to https://magento.stackexchange.com/a/357508/105860, docker-magento's composer call could benefit from this config call first, which would allow Magento 2's < 2.4.4 to keep installing without asking questions:

```
composer config --no-plugins allow-plugins.magento/magento-composer-installer true
composer config --no-plugins allow-plugins.magento/inventory-composer-installer true
composer config --no-plugins allow-plugins.laminas/laminas-dependency-plugin true
```
Would you see this fit for adding the docker-magento one-liner ?

**Comment by markshust** [comment]
@myselfhimself absolutely, thanks so much for the lines! That made it easy for me. This shouldn't adversely affect any version of M2.

Can you confirm that it works now?

**Comment by myselfhimself** [comment]
@markshust Thanks for that patching!

Those lines should ideally be run before the composer create-project downloading call in the `bin/download` script: https://github.com/markshust/docker-magento/blob/f81cfc4d35574a4a7eb0aa61a9cb94fe0e2a450d/compose/bin/download#L24
while you have put them in the `bin/setup` script, which runs after `bin/download` [in the onlinesetup script](https://github.com/markshust/docker-magento/blob/f81cfc4d35574a4a7eb0aa61a9cb94fe0e2a450d/lib/onelinesetup#L11)

I will test your PR tomorrow anyways, I may be mistaken in that too-late placement assumption.

**Comment by myselfhimself** [comment]
Just proposed a PR will the 3 lines located in `bin/download` instead before the `composer create-project` line

#741 seems to a duplicate of this. Asked submitter to try out that PR. Will try it out too on a faster computer...

**Comment by myselfhimself** [comment]
@markshust The PR needs your approval to run Github Actions tests

**Comment by hostep** [comment]
You'll probably best add the `--no-install` flag to the `create-project` command, after that add the allowed plugins, and finally call `composer install`

I think in the way suggested with https://github.com/myselfhimself/docker-magento/commit/6a2fb7e0cc3d959ce7d5a383a27197ba48687d62, it won't work.

Also see https://github.com/composer/composer/issues/10928#issuecomment-1181534484

I would strongly suggest to only apply these changes for Magento versions >= 2.3.7 and < 2.4.4 if that's possible, reasons:
- Magento 2.3.7 was the first version to support composer version 2 and this allowed plugins is only part of composer v2 and not composer v1
- Magento 2.4.4 comes with this allowed-plugins configuration built-in. And those plugins might change again in the future, so applying those for future versions sounds like a bad idea

**Comment by myselfhimself** [comment]
@hostep Thank you, [I improved the PR ](https://github.com/markshust/docker-magento/pull/742/files) with your 3-steps suggestion (composer create project with no-install flag, add 3 allowances for magento-related dependencies, composer install fully)

We could either integrate the >= 2.3.7 to < 2.4.4 filter in that PR or a new one, once the current PR is merge.

**Comment by myselfhimself** [comment]
To me the PR is OK, tested manually locally (using the github CLI tool named `gh`) without the >= 2.3.7 to < 2.4.4 filter yet:
```
gh repo clone markshust/docker-magento docker-magento73126072000_b
gh pr checkout 742
cd docker-magento73126072000_b
rm -rf .git
bash lib/template
#paste https://github.com/myselfhimself/docker-magento/blob/patch-1/compose/bin/download 's contents into bin/download
bin/download 2.4.3-p2
bin/setup magento.test

# open https://magento.test
```

**Comment by jsjsvsvs** [comment]
Thanks or your response @myselfhimself 

I'm getting an error trying to checkout 742

❯ gh pr checkout 742
could not determine base repo: fatal: not a git repository (or any of the parent directories): .git
/usr/bin/git: exit status 128

Also, getting an error (without checkout?) while running bin/setup 


chmod: cannot access 'bin/magento': No such file or directory

**Comment by myselfhimself** [comment]
Apologies, I just added a `cd` line after the `gh clone`  to fix your first error..

**Comment by jsjsvsvs** [comment]
I believe I may be getting the same error (but there might be a step that is incorrect)


docker-magento73126072000_b on  master [?] took 2s
❯ #paste https://github.com/myselfhimself/docker-magento/blob/patch-1/compose/bin/download 's contents into bin/download

docker-magento73126072000_b on  master [?]
❯ bin/download 2.4.3-p2
[+] Running 7/7
 ⠿ Container docker-magento73126072000_b-db-1             Started                                                                                                                                                                        0.8s
 ⠿ Container docker-magento73126072000_b-redis-1          Started                                                                                                                                                                        2.0s
 ⠿ Container docker-magento73126072000_b-rabbitmq-1       Started                                                                                                                                                                        1.9s
 ⠿ Container docker-magento73126072000_b-app-1            Started                                                                                                                                                                        2.1s
 ⠿ Container docker-magento73126072000_b-elasticsearch-1  Started                                                                                                                                                                        0.8s
 ⠿ Container docker-magento73126072000_b-mailcatcher-1    Started                                                                                                                                                                        1.9s
 ⠿ Container docker-magento73126072000_b-phpfpm-1         Started                                                                                                                                                                        2.0s
Composer auth has already been set up.
Fixing filesystem ownerships...
Filesystem ownerships fixed.
magento/magento-composer-installer contains a Composer plugin which is blocked by your allow-plugins config. You may add it to the list if you consider it safe. See https://getcomposer.org/allow-plugins
You can run "composer config --no-plugins allow-plugins.magento/magento-composer-installer [true|false]" to enable it (true) or keep it disabled and suppress this warning (false)
magento/inventory-composer-installer contains a Composer plugin which is blocked by your allow-plugins config. You may add it to the list if you consider it safe. See https://getcomposer.org/allow-plugins
You can run "composer config --no-plugins allow-plugins.magento/inventory-composer-installer [true|false]" to enable it (true) or keep it disabled and suppress this warning (false)
laminas/laminas-dependency-plugin contains a Composer plugin which is blocked by your allow-plugins config. You may add it to the list if you consider it safe. See https://getcomposer.org/allow-plugins
You can run "composer config --no-plugins allow-plugins.laminas/laminas-dependency-plugin [true|false]" to enable it (true) or keep it disabled and suppress this warning (false)
dealerdirect/phpcodesniffer-composer-installer contains a Composer plugin which is blocked by your allow-plugins config. You may add it to the list if you consider it safe. See https://getcomposer.org/allow-plugins
You can run "composer config --no-plugins allow-plugins.dealerdirect/phpcodesniffer-composer-installer [true|false]" to enable it (true) or keep it disabled and suppress this warning (false)
magento/composer-dependency-version-audit-plugin contains a Composer plugin which is blocked by your allow-plugins config. You may add it to the list if you consider it safe. See https://getcomposer.org/allow-plugins
You can run "composer config --no-plugins allow-plugins.magento/composer-dependency-version-audit-plugin [true|false]" to enable it (true) or keep it disabled and suppress this warning (false)
magento/composer-root-update-plugin contains a Composer plugin which is blocked by your allow-plugins config. You may add it to the list if you consider it safe. See https://getcomposer.org/allow-plugins
You can run "composer config --no-plugins allow-plugins.magento/composer-root-update-plugin [true|false]" to enable it (true) or keep it disabled and suppress this warning (false)
Creating a "magento/project-community-edition=2.4.3-p2" project at "./"


  [InvalidArgumentException]
  Project directory "/var/www/html/." is not empty.


create-project [-s|--stability STABILITY] [--prefer-source] [--prefer-dist] [--prefer-install PREFER-INSTALL] [--repository REPOSITORY] [--repository-url REPOSITORY-URL] [--add-repository] [--dev] [--no-dev] [--no-custom-installers] [--no-scripts] [--no-progress] [--no-secure-http] [--keep-vcs] [--remove-vcs] [--no-install] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--ask] [--] [<package>] [<directory>] [<version>]


docker-magento73126072000_b on  master [?] took 21s
❯ bin/setup magento.test
[+] Running 7/7
 ⠿ Container docker-magento73126072000_b-db-1             Stopped                                                                                                                                                                        1.9s
 ⠿ Container docker-magento73126072000_b-rabbitmq-1       Stopped                                                                                                                                                                        6.6s
 ⠿ Container docker-magento73126072000_b-redis-1          Stopped                                                                                                                                                                        0.4s
 ⠿ Container docker-magento73126072000_b-mailcatcher-1    Stopped                                                                                                                                                                        0.4s
 ⠿ Container docker-magento73126072000_b-elasticsearch-1  Stopped                                                                                                                                                                        1.8s
 ⠿ Container docker-magento73126072000_b-phpfpm-1         Stopped                                                                                                                                                                        0.5s
 ⠿ Container docker-magento73126072000_b-app-1            Stopped                                                                                                                                                                        1.2s
[+] Running 7/7
 ⠿ Container docker-magento73126072000_b-app-1            Started                                                                                                                                                                        1.2s
 ⠿ Container docker-magento73126072000_b-redis-1          Started                                                                                                                                                                        1.1s
 ⠿ Container docker-magento73126072000_b-rabbitmq-1       Started                                                                                                                                                                        0.4s
 ⠿ Container docker-magento73126072000_b-mailcatcher-1    Started                                                                                                                                                                        1.2s
 ⠿ Container docker-magento73126072000_b-phpfpm-1         Started                                                                                                                                                                        1.1s
 ⠿ Container docker-magento73126072000_b-elasticsearch-1  Started                                                                                                                                                                        1.1s
 ⠿ Container docker-magento73126072000_b-db-1             Started                                                                                                                                                                        1.0s
chmod: cannot access 'bin/magento': No such file or directory

**Comment by myselfhimself** [comment]
You should stop your previously created running docker images between attempts (so, basically now) with:
`bin/stopall`

Also each new attempt would need a new directory..
So:
` gh repo clone markshust/docker-magento docker-magento73126072000_b # put a different directory name here`
and 
`cd docker-magento73126072000_b #same directory name as before `
or just 
`cd $_`

**Comment by myselfhimself** [comment]
@hostep I added version filtering to the PR as you recommended. Needs testing.

**Comment by anilacikgoz** [comment]
I resolved this issue with adding these lines the on bin/download file;
```
bin/clinotty composer create-project --repository=https://repo.magento.com/ magento/project-"${EDITION}"-edition="${VERSION}"  . --no-install


echo "Adding Magento modules to Composer allow-plugins directive..."
bin/clinotty composer config --no-plugins allow-plugins.magento/magento-composer-installer true
bin/clinotty composer config --no-plugins allow-plugins.magento/inventory-composer-installer true
bin/clinotty composer config --no-plugins allow-plugins.laminas/laminas-dependency-plugin true

bin/clinotty composer update
```

**Comment by nhapq** [comment]
I tried all of your methods but still got this error: 

`Creating a "magento/project-community-edition=2.4.4" project at "./"

                                                     
  [InvalidArgumentException]                         
  Project directory "/var/www/html/." is not empty.  
                                                     

create-project [-s|--stability STABILITY] [--prefer-source] [--prefer-dist] [--prefer-install PREFER-INSTALL] [--repository REPOSITORY] [--repository-url REPOSITORY-URL] [--add-repository] [--dev] [--no-dev] [--no-custom-installers] [--no-scripts] [--no-progress] [--no-secure-http] [--keep-vcs] [--remove-vcs] [--no-install] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--ask] [--] [<package>] [<directory>] [<version>]

Composer could not find a composer.json file in /var/www/html
To initialize a project, please create a composer.json file. See https://getcomposer.org/basic-usage`

I have reinstalled from fresh many times. System: Debian 8 GB ram.

**Comment by anilacikgoz** [comment]
@nhapq I think you should clear all content in your container(/var/www/html). 
Use bin/bash command and check /var/www/html directory whether empty or not.
After that you can try again. 
With standard usage, Composer create-project command needs  an empty project directory.

**Comment by ChristianTestUser** [comment]
Is Magento version 2.3.4 still supported by you?

I have seen that support for php 7.3 has been removed.

**Comment by markshust** [comment]
I don't necessarily support specific versions of Magento anymore, but moreso the underlying PHP versions. 7.3 reached EOL a while ago, so that support was dropped. 7.4 support will be dropped at the end of November, leaving only 8.1 (and probably 8.2 at that time).

@nhapq any error "Project directory "/var/www/html/." is not empty." means the directory is not empty, so it needs to be fully purged before trying to re-execute the install as @anilacikgoz pointed out. You can do this with `bin/removeall`, backing out of the directory, removing the entire project, then starting over.

I believe this issue about composer allow-plugins was already fixed a while ago? If not, please let me know.

**Comment by omergilad** [comment]
Also experiencing this issue when installing 2.4.3-p1 community,
having to use the manual workarounds described here.

I will add those to my scripts, which i keep in version control after the initial docker-compose template clone.

Any plan to fix it in this project?

**Comment by markshust** [comment]
Magento 2.4.3 officially hits end-of-life tomorrow, so I'll be closing this out as "won't fix".

PHP 7 image will also get deprecated (meaning you can use it, but I won't be taking any updates in).

When something reaches end-of-life, it's dead and needs to stop being used ASAP! I don't plan on supporting anything that is already buried in a coffin 😁

**Comment by tbunitrade** [comment]
> #paste https://github.com/myselfhimself/docker-magento/blob/patch-1/compose/bin/download 's contents into bin/download

Not helped :( 
magento % bin/download 2.4.3-p2
[+] Running 8/8
 ⠿ Network magento_default            Created                                                                       0.1s
 ⠿ Container magento-rabbitmq-1       Healthy                                                                       7.3s
 ⠿ Container magento-elasticsearch-1  Healthy                                                                      17.8s
 ⠿ Container magento-redis-1          Healthy                                                                       7.8s
 ⠿ Container magento-mailcatcher-1    Started                                                                       0.8s
 ⠿ Container magento-db-1             Healthy                                                                       6.3s
 ⠿ Container magento-phpfpm-1         Started                                                                      18.4s
 ⠿ Container magento-app-1            Started                                                                      18.9s
Composer auth has already been set up.
Fixing filesystem ownerships...
Filesystem ownerships fixed.
Creating a "magento/project-community-edition=2.4.3-p2" project at "./"

                                                     
  [InvalidArgumentException]                         
  Project directory "/var/www/html/." is not empty.

**Comment by tbunitrade** [comment]
> curl -s https://raw.githubusercontent.com/markshust/docker-magento/master/lib/onelinesetup | bash -s -- magento.test 2.4.3-p2

Composer auth has already been set up.
Fixing filesystem ownerships...
Filesystem ownerships fixed.
Creating a "magento/project-community-edition=2.4.3-p2" project at "./"

                                                     
  [InvalidArgumentException]                         
  Project directory "/var/www/html/." is not empty.

**Comment by markshust** [comment]
Everything is working properly, so unable to replicate.

Your error is leading to artifacts being left over from a preexisting install attempt. Please see [markshust/docker-magento#install-fails-because-project-directory-is-not-empty](https://github.com/markshust/docker-magento/#install-fails-because-project-directory-is-not-empty)

**Comment by tbunitrade** [comment]
@markshust  Yes I do not use credentials from this portal https://marketplace.magento.com/
I update here credentials `  cd ~/.composer auth.json`
Than I delete all `bin/removeall` all setup go well, but I still has this issue  chmod cannot access 'bin/magento': No such file or directory


487/487 [============================] 100%
117 package suggestions were added by new dependencies, use `composer suggest` to see details.
Package container-interop/container-interop is abandoned, you should avoid using it. Use psr/container instead.
Package laminas/laminas-console is abandoned, you should avoid using it. Use laminas/laminas-cli instead.
Package csharpru/vault-php-guzzle6-transport is abandoned, you should avoid using it. No replacement was suggested.
Package hoa/consistency is abandoned, you should avoid using it. No replacement was suggested.
Package hoa/console is abandoned, you should avoid using it. No replacement was suggested.
Package hoa/event is abandoned, you should avoid using it. No replacement was suggested.
Package hoa/exception is abandoned, you should avoid using it. No replacement was suggested.
Package hoa/file is abandoned, you should avoid using it. No replacement was suggested.
Package hoa/iterator is abandoned, you should avoid using it. No replacement was suggested.
Package hoa/protocol is abandoned, you should avoid using it. No replacement was suggested.
Package hoa/stream is abandoned, you should avoid using it. No replacement was suggested.
Package hoa/ustring is abandoned, you should avoid using it. No replacement was suggested.
Package php-cs-fixer/diff is abandoned, you should avoid using it. No replacement was suggested.
Package phpunit/php-token-stream is abandoned, you should avoid using it. No replacement was suggested.
Package sebastian/finder-facade is abandoned, you should avoid using it. No replacement was suggested.
Package theseer/fdomdocument is abandoned, you should avoid using it. No replacement was suggested.
Generating autoload files
102 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
[+] Running 7/7
 ⠿ Container magento-app-1            Stopped                                                                       1.6s
 ⠿ Container magento-mailcatcher-1    Stopped                                                                       1.6s
 ⠿ Container magento-phpfpm-1         Stopped                                                                       0.2s
 ⠿ Container magento-rabbitmq-1       Stopped                                                                       6.4s
 ⠿ Container magento-redis-1          Stopped                                                                       0.3s
 ⠿ Container magento-elasticsearch-1  Stopped                                                                       5.5s
 ⠿ Container magento-db-1             Stopped                                                                       1.5s
[+] Running 7/7
 ⠿ Container magento-rabbitmq-1       Healthy                                                                       7.0s
 ⠿ Container magento-elasticsearch-1  Healthy                                                                      16.5s
 ⠿ Container magento-redis-1          Healthy                                                                       6.0s
 ⠿ Container magento-db-1             Healthy                                                                       6.0s
 ⠿ Container magento-mailcatcher-1    Started                                                                       1.0s
 ⠿ Container magento-phpfpm-1         Started                                                                      17.4s
 ⠿ Container magento-app-1            Started                                                                      17.9s
chmod: cannot access 'bin/magento': No such file or directory

@markshust What I missed?

**Comment by tbunitrade** [comment]
osony@C02F45AFMD6P magento % bin/setup
[+] Running 7/7
 ⠿ Container magento-app-1            Stopped                                                                       2.7s
 ⠿ Container magento-mailcatcher-1    Stopped                                                                       1.0s
 ⠿ Container magento-phpfpm-1         Stopped                                                                       0.3s
 ⠿ Container magento-elasticsearch-1  Stopped                                                                       2.2s
 ⠿ Container magento-rabbitmq-1       Stopped                                                                       6.8s
 ⠿ Container magento-db-1             Stopped                                                                       2.6s
 ⠿ Container magento-redis-1          Stopped                                                                       0.3s
[+] Running 7/7
 ⠿ Container magento-redis-1          Healthy                                                                       5.9s
 ⠿ Container magento-mailcatcher-1    Started                                                                       0.7s
 ⠿ Container magento-rabbitmq-1       Healthy                                                                       6.9s
 ⠿ Container magento-elasticsearch-1  Healthy                                                                      16.5s
 ⠿ Container magento-db-1             Healthy                                                                       6.0s
 ⠿ Container magento-phpfpm-1         Started                                                                      17.4s
 ⠿ Container magento-app-1            Started                                                                      17.9s
chmod: cannot access 'bin/magento': No such file or directory

**Comment by markshust** [comment]
Your install probably failed at some point. Please see https://github.com/markshust/docker-magento#install-fails-because-project-directory-is-not-empty
